aiida/backends/djsite/db/migrations/0024_dblog_update.py

```diff
@@ -6,39 +6,216 @@
+
+node_prefix = 'node.'
+leg_workflow_prefix = 'aiida.workflows.user.'
+
+
+def get_legacy_workflow_log_no(dblog_model):
+    """ Get the number of the log records that correspond to legacy workflows """
+    return dblog_model.objects.filter(objname__startswith=leg_workflow_prefix).count()
+
+
+def get_unknown_entity_log_no(dblog_model):
+    """ Get the number of the log records that correspond to unknown entities """
+    return dblog_model.objects.exclude(objname__startswith=node_prefix).exclude(
+        objname__startswith=leg_workflow_prefix).count()
+
+
+def get_serialized_legacy_workflow_logs(dblog_model):
+    """ Get the serialized log records that correspond to legacy workflows """
+    from aiida.backends.sqlalchemy.utils import dumps_json
+    queryset = dblog_model.objects.filter(objname__startswith=leg_workflow_prefix).values(*values_to_export)
+    return dumps_json(list(queryset))
+
+
+def get_serialized_unknown_entity_logs(dblog_model):
+    """ Get the serialized log records that correspond to unknown entities """
+    from aiida.backends.sqlalchemy.utils import dumps_json
+    queryset = dblog_model.objects.exclude(objname__startswith=node_prefix).exclude(
+        objname__startswith=leg_workflow_prefix).values(*values_to_export)
+    return dumps_json(list(queryset))
+
+
+def set_new_uuid(apps, _):
+    """
+    Set new UUIDs for all logs
+    """
+    from django.db import transaction
+
+    DbLog = apps.get_model('db', 'DbLog')
+    with transaction.atomic():
+        for log in DbLog.objects.all():
+            log.uuid = aiida.common.utils.get_new_uuid()
+            log.save(update_fields=['uuid'])
+
+
+def export_and_clean_workflow_logs(apps, _):
+    """
+    Export the logs records that correspond to legacy workflows and to unknown entities.
+    """
+    from django.db import transaction
+    from tempfile import NamedTemporaryFile
+
+    DbLog = apps.get_model('db', 'DbLog')
+    lwf_no = DbLog.objects.filter(objname__startswith=leg_workflow_prefix).count()
+    other_no = DbLog.objects.exclude(objname__startswith=node_prefix).exclude(
+        objname__startswith=leg_workflow_prefix).count()
+
+    # If there are no legacy workflow log records or log records of an unknown entity
+    if lwf_no == 0 and other_no == 0:
+        return
+
+    if not settings.TESTING_MODE:
+        click.echo('We found {} log records that correspond to legacy workflows and {} log records to correspond '
+                   'to an unknown entity.'.format(lwf_no, other_no))
+        click.echo(
+            'These records will be removed from the database and exported to JSON files to the current directory).')
+        proceed = click.confirm('Would you like to proceed?', default=True)
+        if not proceed:
+            return
+
+    delete_on_close = False
+    if settings.TESTING_MODE:
+        delete_on_close = True
+
+    # Exporting the legacy workflow log records
+    if lwf_no != 0:
+        with transaction.atomic():
```

Comment:
I added it because I wasn't sure that Django runs every migration in a transaction (very likely yes) and I wanted to identify sub-blocks of code that should be atomic.

But I assume that the whole migration is run as one transaction and I can remove the sub-transactions.